### PR TITLE
build(gh): move build to github actions

### PR DESCRIPTION
**Pull request by schmidtk**
travis-ci.org is no longer building for free, so we'll move to GitHub Actions instead.

- The Yarn cache step saves the cache based on a hash of the `yarn.lock`, so that was added to the repo.
- Commitlint is a separate action from the build, for convenience. This follows the pattern established in the build projects.
- Scripts designed for Travis were removed and replaced within `build.yml`. The primary difference is that the surge deployment only happens on the `master` build now, but I don't believe we need to keep track of deployments for anything else.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Build
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (wallw-bits) [inline .github/workflows/build.yml:7]: We've had a couple of PRs to `release`, shouldn't that get handled here as well?

File: .github/workflows/commitlint.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Lint Commit Messages
+
+on:
+  pull_request:
+    branches: [ master ]
```
Comment (wallw-bits) [inline .github/workflows/commitlint.yml:5]: Same here.

**Review by wallw-bits — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (schmidtk) [reply .github/workflows/build.yml]: Yes, thank you. Fixed.

File: .github/workflows/commitlint.yml (hunk 2 above)
Comment (schmidtk) [reply .github/workflows/commitlint.yml]: Also fixed.

**Review by wallw-bits — APPROVED** [review]

**Pull request merged**